homeassistant/components/script/__init__.py

```diff
@@ -264,16 +264,15 @@ async def _async_process_config(hass, config, component) -> bool:
                 raw_config = cast(ScriptConfig, config_block).raw_config
 
             entities.append(
-                ScriptEntity(
-                    hass, object_id, config_block, raw_config, raw_blueprint_inputs
-                )
+                ScriptEntity(hass, key, config_block, raw_config, raw_blueprint_inputs)
             )
 
     await component.async_add_entities(entities)
 
     async def service_handler(service: ServiceCall) -> None:
         """Execute a service call to script.<script name>."""
-        entity_id = ENTITY_ID_FORMAT.format(service.service)
+        entity_registry = er.async_get(hass)
+        entity_id = entity_registry.async_get_entity_id(DOMAIN, DOMAIN, service.service)
```

Comment:
This is the actual bugfix: The service is not same as the 2nd half of the `entity_id` if the `entity_id` has been overridden.